### DISCUSSION TITLE
SOLVED! Essentials custom join message via config not working/others

### DISCUSSION THREAD

**Discussion by koalajaden** [EssentialsX/Essentials#4061] [Q&A] [answered]
Hi, I recently downloaded EssentialsX and saw the option to change/add a custom join message here:

![customjoinm1](https://user-images.githubusercontent.com/80790666/111380888-53cd6080-867b-11eb-8cda-cf8f40708fbe.png)

Upon joining, it doesn't seem like the message is working. I apologize in advance if I'm being a dunce and this is a waste of time, I'm a bit new to Java coding and am trying to learn as I go. 

The AFK-playerlist also does not work, set up as above: 

![customjoinm2](https://user-images.githubusercontent.com/80790666/111381115-a575eb00-867b-11eb-826a-af71fc1fbeb9.png)

**Comment by triagonal** [comment]
For your first problem, make sure that you haven't accidentally negated `bukkit.broadcast.user` by negating `*` or `bukkit.*` for your default group. If it still doesn't work after you've done that, there's likely another plugin handling your join/leave messages.

For your second, make sure you have `change-displayname` and `change-playerlist` set to true in your config.

I'd recommend joining the MOSS support discord for faster/easier support, since we usually require a number of files to diagnose an issue and can help you debug step-by-step. Join here: https://discord.gg/PHpuzZS

**Reply by koalajaden** [reply]
Okay, so I tried checking permissions and it seems that bukkit was never negated.. 

I did check if change-displayname was set to true, as the other one, and they were also already set to true. 

I did just add luckperms to make sure it wasn't some weird permission plugin I didn't know about, and added bukkit.broadcast.user to the default group.. No change there. Hmm..

Should I try and post it inside the discord?

I could also send you some photos of my logs/console if needed.

**Comment by koalajaden** [comment, marked as answer]
Bing bing! Found the bug:

After much diddling, turns out it was EssentialsX, unbeknownst to me, switching on over to Paper optimization.

While switching the symbols to & may have worked in a past version, it sends an error through Essentials.. Not sure when that changed, but using the section sign (§) as a replacement for (&) seems to be all around good practice for developing in minecraft-java plugins/add-ons for color chat in bukkit.

Switched to latest paper 1.16.5 build, updated my EssentialsX files (EssentialsX to latest dev build 2.19, with EssentialsXChat andEssentials XSpawn to 2.19 as well) and the join messages are working like a charm!..

But now the changing-of-op-names isn't.. Though I'm sure it wouldn't have worked via the 1.16.5 spigot version I was running earlier either. Time to ask around!.. Heh.
